### PR TITLE
Add upstream service address to response headers based on Envoy metadata

### PR DESCRIPTION
**Pull request by franek1709**

### CODE REVIEW

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoyEgressRoutesFactory.kt [hunk 1]
```diff
@@ -88,7 +97,10 @@ internal class EnvoyEgressRoutesFactory(
                     )
                 }
             }
-            .build()
+        if (addUpstreamAddressHeader)
```
Comment (pzmi) [inline envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoyEgressRoutesFactory.kt]: Little style nitpick, in a line 116 we use curly braces in single line if statement. I thought we could configure detekt/ktlint rule in order to maintain consistent style.

**Comment by pzmi** [comment]
What about functional tests? Should we add one in envoy-control-tests? I'm not even sure if we can do it with our current test infrastructure.

**Comment by slonka** [comment]
> What about functional tests? Should we add one in envoy-control-tests? I'm not even sure if we can do it with our current test infrastructure.

That should not be a problem, just add the appropriate config option to config_ads / config_xds yaml file and add a test case with a header check.

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoyEgressRoutesFactory.kt (hunk 1 above)
Comment (franek1709) [reply envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoyEgressRoutesFactory.kt]: Done. I had to do some additional changes after changing detekt config - there was no curly braces in few if statements in our existing code.

**Comment by franek1709** [comment]
> > What about functional tests? Should we add one in envoy-control-tests? I'm not even sure if we can do it with our current test infrastructure.
> 
> That should not be a problem, just add the appropriate config option to config_ads / config_xds yaml file and add a test case with a header check.

Done

**Pull request merged**